### DISCUSSION TITLE
Schema help and suggestion

### DISCUSSION THREAD

**Discussion by ericsonluciano** [prisma/prisma#5223] [Q&A] [answered]
Hello, I need a piece of advice I have 2 user roles type an admin and school the goal admin can create multiple schools and Students but the School creates one school with multiple student. Student have 1:1 of location but what is a good schema for this here my initial schema
```prisma
enum UserRole {
ADMIN
SCHOOL
}

model User {
  id               String      @id @default(cuid())
  email         String      @unique
  school       School
  students   Students[]
  role            UserRole
}

model School {
  id               String      @id @default(cuid())
  name         String      @unique
  students    Student[]
  authorId    String       @unique
  author       User      @relation(fields: [authorId], references: [id])
}

model Student {
  id                String      @id @default(cuid())
  firstName  String
  lastName   String
  location      Location?

  schoolId    String
  school       School           @relation(fields: [schoolId], references: [id])
  authorId    String       @unique
  author       User      @relation(fields: [authorId], references: [id])
}

model Location {
 city              String
 country       String
 fullAddress String

  studentId    String       @unique
  student       Student      @relation(fields: [studentId], references: [id])
}
```

Thanks for the advice and have a great day!

**Comment by 2color** [comment, marked as answer]
Hey @ericsonluciano,

Your data model should work for implementing the authorization logic. 

### Notes/Suggestions
- I'd make the school field in the `User` model optional so that users with the `ADMIN` role don't need to be associated with a specific school.
- One thing that is not clear is why you have the relation between `Student:User` with the `authorId` field in the `Student` model? Is that just to track which `User` created the `Student`? 
Since you know the School related to the User (1:1), I'm not sure that you need the `authorId` and `author` fields in the `Student` model.

Otherwise the schema looks fine. Beware, that while there's nothing wrong with 1:1 relations, you could simplify the schema by storing the location fields in the `Student` model and removing the `Location` model

You could also further optimize the way you store cities and countries by using a many-to-many between Location and a City/Country table. That way you will not be storing a copy of the city name for every instance of it. However, this might be premature optimization.

**Reply by ericsonluciano** [reply]
Hey thanks for the suggestion

- for the `Student:User` I removed the relationship with this since it's also connected with the `School` Model
- on the `Location` i decided to make it into many-to-many since it's also a good option for the filters

Thanks again for the help!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
